### DISCUSSION TITLE
Enhancement: Google Analytics or Google Tag Manager support

### DISCUSSION THREAD

**Discussion by virtuman** [danny-avila/LibreChat#2430] [Feature Requests & Suggestions]
### What features would you like to see added?

Would be nice to add some sort of analytics. Probably adding google tag manager is the best way, because it would allow for any custom setup.

### More details

environment based variable, to specify GOOGLE_TAG_MANAGER_ID should include necessary javascript in the page. this would allow to enable any other additional tracking and customization features via google tag manager.

### Which components are impacted by your request?

UI

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by DenisPalnitsky** [comment]
That would be a useful feature. 
I'm not sure that Google Tag Manager is the best option as it requires fair amount of  manual configuration to setup and maintain. However I'm not an expert in this area and could miss something. 

I suggest to use google analytics if there are no better options

**Comment by DenisPalnitsky** [comment]
I guess, Google Tag Manager is better then nothing. Please check https://github.com/danny-avila/LibreChat/pull/2469

**Comment by leondape** [comment]
How about a .env or librechat.yml setting that adds a tracking code like that from matomo to the footer or so?

**Reply by DenisPalnitsky** [reply]
It's already added https://www.librechat.ai/docs/configuration/dotenv#google-tag-manager
